pegasus/sites.v3/code.org/public/private/edit-csp-table/splat.haml

```diff
@@ -4,91 +4,133 @@
-    %tr
-      %td
-        #{record[:row_id]}
-      %td
-        %textarea{id: "#{record[:row_id]}-textarea", name:"record", value:"#{record[:value]}", cols:"80", rows:"2"}= record[:value]
-      %td.button-column
-        %button{id:"#{record[:row_id]}-save"}
-          Save
-        %button{id:"#{record[:row_id]}-delete"}
-          Delete
-  %tr
-    %td (new)
-    %td
-      %textarea{id: "new-record-textarea", name:'record', cols:"80", rows:"2"}= jsonExample
-    %td
-      %button#new-record-add Add
+%div{"ng-app" => "recordsApp", "ng-view" => ""}
 
 :javascript
-  function processResponse(data) {
-    window.location.reload()
-  }
-  function processError(data) {
-    window.alert(data.status + '\n' + data.responseJSON);
-  }
-  function saveRecordData(url, recordData) {
-    try {
-      JSON.parse(recordData); // make sure it parses
-      $.ajax({
-        url: url,
-        type: 'post',
-        data: recordData,
-        processData: false,
-        contentType: 'application/json; charset=utf-8'
-      }).done(processResponse).fail(processError);
-    } catch (e) {
-      window.alert(
-          'Error parsing JSON:\n' +
-          '\tinput: ' + recordData + '\n' +
-          '\terror: "' + e.message + '"\n\n' +
-          'example of valid JSON: \n\t#{jsonExample}');
+  'use strict';
+
+  // Declare app level module which depends on filters, and services
+  angular.module('recordsApp', [
+    'ngRoute',
+    'ngResource',
+    'recordsApp.controllers',
+    'recordsApp.services'
+  ]).config(['$routeProvider', function($routeProvider) {
+    $routeProvider.when('/',
+                        {templateUrl: '/private/edit-csp-table/records',
+                         controller: 'RecordsController'});
+    $routeProvider.otherwise({redirectTo: '/'});
+  }]);
+
+  // SERVICES
+  var services = angular.module('recordsApp.services', []).
+      value('version', '0.1');
+
+  services.factory('recordsService', ['$resource',
+    function($resource) {
+      var Record = $resource("#{api_base_url}/:recordId", {recordId:'@id'}, {
+        // default methods: see https://code.angularjs.org/1.2.21/docs/api/ngResource/service/$resource
+        //  'get':    {method: 'GET'},
+        //  'save':   {method: 'POST'},
+        //  'query':  {method: 'GET', isArray:true},
+        //  'remove': {method: 'DELETE'},
+        //  'delete': {method: 'DELETE'} // don't use this because it doesn't work in IE9
+      });
+
+      return Record;
+    }]);
+
+  // CONTROLLERS
+
+  var controllers = angular.module('recordsApp.controllers', []).
+      value('version', '0.1');
+
+  controllers.controller('RecordsController', ['$scope', '$route', '$routeParams', '$location', '$window', 'recordsService',
+      function($scope, $route, $routeParams, $location, $window, recordsService) {
+    $scope.records = recordsService.query();
+    $scope.newColumnName = '';
+    $scope.newRecord = new recordsService();
+    $scope.tableName = "#{table_name}";
```

Comment:
Yes, I believe it is safe to cache the page because pages are cached by URL and these attributes are in (or derived from) the url.